native/core/src/execution/operators/parquet_writer.rs

```diff
@@ -197,11 +224,18 @@ impl ExecutionPlan for ParquetWriterExec {
             Arc::clone(&input_schema)
         };
 
+        // Determine the write path (work_dir for temp files, or output_path for direct write)
```

Comment:
Do we still need to keep the behavior of writing directly to `output_path`? This seems risky, because we don’t have any failure cleanup in place.

Reply:
Good point. This should be removed now.